download/all.md

```diff
@@ -1,8 +1,26 @@
 ---
-layout: alldownloadspage
-title: Download Previous Versions
-development_version: 2.11.0-M5
+layout: page
+title: All Available Versions
 ---
 
-<!-- This page should be auto-generated -->
+<p>This page contains a comprehensive archive of previous Scala releases.</p>
 
+{% for top in (0..3) reversed %}
+  {% for major in (0..20) reversed %}
+    {% assign possibleVersionShort = top | append:'.' | append:major %}
+    {% assign sz = possibleVersionShort | size %}
+    {% if 3 == sz %}
+      {% assign possibleVersion = possibleVersionShort | append:'.' %}
+    {% else %}
+      {% assign possibleVersion = possibleVersionShort %}
+    {% endif %}
+    {% for page in site.categories.download %}
+      {% assign releaseVersion = page.release_version | truncate:4, '' %}
+      {% if releaseVersion == possibleVersion %}
+        <div>
+          <a href="{{ page.url }}">{{ page.title }}</a>
+        </div>
```

Comment:
This HTML needs to start in the first column, instead it's being shown as source code:

```
<div>
  <a href="/download/2.12.0.html">Scala 2.12.0</a>
</div>
  
  
  
<div>
  <a href="/download/2.12.0-RC2.html">Scala 2.12.0-RC2</a>
</div>
  
...
```